source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,147 @@
+==================
+Tuning Connections
+==================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. _connection-storms-java:
+
+Overview
+--------
+
+In this guide, you can learn about tuning connections to prevent **connection storms**,
+what connection storms are, and common causes.
+
+A **connection storm** is an event in which application servers
+start an increasing number of connection requests, but the host server
+CPU is unable to schedule them.
+
+.. example::
+
+   Consider an application that uses MongoDB, and is served from an
+   auto scaling service. A sudden spike in traffic could cause the scaling
+   service to stand up many new application instances. This sudden burst
+   in application instances attempting to establish connections to MongoDB
+   is one example of a connection storm.
+
+Common causes for connection storms in MongoDB include:
+
+- Incorrect application behavior
+- Network latency
+- Insufficient hardware for the workload
+- Failover events
+
+In MongoDB, connection storms can have a multiple effects. For example:
+
+- The high volume of connection requests leads to latency in processing.
+- Replication connections between nodes are blocked or slowed, so replication
+  operations cannot proceed.
+- Administrative connections to the database are blocked, preventing
+  a user from logging in.
+- Monitoring connections to the database are blocked, degrading visibility
+  into the problem.
+- CPU or memory oversubscription due to connection management can lead
+  to a ``mongod`` crashing and restarting. This extra workload on the server
+  can be exclusively attributed to connection handling.
+- In Atlas, connection requests can exceed instance class limits, causing
+  refusal of new connections.
+
+Avoid Connection Storms
+-----------------------
+
+A combination of settings and methods can help to mitigate connection storms.
+These include:
```

Comment:
maxPoolSize would be another good value to mention. Lower values can also help reduce storms simply by reducing the amount of connections, and, like maxConnecting, can actually lead to increased throughput for certain workloads. The driver default of 100 is often too high, so including that info here would be especially helpful.

See https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing for some more info on this.